source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -57,12 +61,35 @@ bool ProcessorState::handleHeadersResponse(const HeadersResponse& response,
 }
 
 bool ProcessorState::handleBodyResponse(const BodyResponse& response) {
-  if (callback_state_ == CallbackState::BufferedBody) {
+  if (callback_state_ == CallbackState::BufferedBodyCallback) {
     ENVOY_LOG(debug, "Applying body response to buffered data");
     modifyBufferedData([this, &response](Buffer::Instance& data) {
       MutationUtils::applyCommonBodyResponse(response, headers_, data);
     });
     headers_ = nullptr;
+    callback_state_ = CallbackState::BodyComplete;
+    message_timer_->disableTimer();
```

Comment:
Why don't we track timeout for trailers?

Reply:
We're just cancelling the timer that was set for the body response. In the call to filter_.sendTrailers() we'll re-enable it for the timers message.